fieldservice_vehicle_stock/models/stock_picking.py

```diff
@@ -11,26 +11,16 @@ class StockPicking(models.Model):
 
     @api.multi
     def action_assign(self):
-        res = {}
         for rec in self:
-            if rec.picking_type_id in (
-                # Vehicle Loading
-                self.env.ref(
-                    'fieldservice_vehicle_stock.'
-                    'picking_type_output_to_vehicle'),
-                # Location Pickup
-                self.env.ref(
-                    'fieldservice_vehicle_stock.'
-                    'picking_type_location_to_vehicle')
-            ):
+            if rec.picking_type_id.require_vehicle_id:
                 if rec.fsm_vehicle_id:
                     picking = \
                         rec.with_context(vehicle_id=rec.fsm_vehicle_id.id)
-                    res = super(StockPicking, picking).action_assign()
+                    return super(StockPicking, picking).action_assign()
```

Comment:
@osi-scampbell This will stop the loop and the logic will not apply to the next records...